dhall/src/Dhall/Binary.hs

```diff
@@ -1178,7 +1178,7 @@ instance Serialise (Expr Void Import) where
     decode = decodeExpressionInternal decodeImport
 
 -- | Encode a Dhall expression as a CBOR-encoded `ByteString`
-encodeExpression :: Expr Void Import -> ByteString
+encodeExpression :: Serialise (Expr Void a) => Expr Void a -> ByteString
```

Comment:
Just a question, probably for @Gabriel439: Why don't we have an instance

```
instance (Serialise a) => Serialise (Expr Void a)
```
, which would enable us to have this type signature here:

```suggestion
encodeExpression :: Serialise a => Expr Void a -> ByteString
```

? Because there's no official instance for `Void`?

Reply:
@sjakobi: It's because of the type of `decodeExpressionInternal`, which is used to power the decoding logic:

```haskell
decodeExpressionInternal :: (Int -> Decoder s a) -> Decoder s (Expr t a)
```

The required `Int` parameter to the `Decoder` for the `a` means that we can't just use the `Decoder` that the `Serialise` instance for `a` would give us.

Reply:
Thanks @Gabriel439! That makes sense! :)

Reply:
@sjakobi: You're welcome! 🙂